File: elixir/bob/2/lib/bob.ex
defmodule Bob do
  @spec question?(binary()) :: boolean()
  def question?(input), do:
    String.at(input, -1) == "?"

  @spec yelling?(binary()) :: boolean()
  def yelling?(input), do:
    String.upcase(input) == input and String.downcase(input) != input

  @spec silence?(any()) :: boolean()
  def silence?(""), do: true
  def silence?(_), do: false

  @spec hey(String.t()) :: String.t()
  def hey(input) do
    trimmed_input = String.trim(input)
    {question, yelling, silence} = {question?(trimmed_input), yelling?(trimmed_input), silence?(trimmed_input)}
    IO.inspect({question, yelling, silence})
    case {question, yelling, silence} do
      {_, _, true} -> "Fine. Be that way!"
      {true, true, _} -> "Calm down, I know what I'm doing!"
      {true, false, _} -> "Sure."
      {false, true, _} -> "Whoa, chill out!"
      {_, _, _} -> "Whatever."
    end
  end
end

File: python/bob/1/bob.py
def response(hey_bob: str):
    hey_bob = hey_bob.strip()
    if hey_bob ==  "":
        return "Fine. Be that way!"
    if hey_bob[-1] == "?" and hey_bob.isupper():
        return "Calm down, I know what I'm doing!"
    if hey_bob[-1] == "?":
        return "Sure."
    if hey_bob.isupper():
        return "Whoa, chill out!"
    return "Whatever."